### DISCUSSION TITLE
Sensor12 - Expanded documentation for ADS1115 Modes

### DISCUSSION THREAD

**Discussion by phinnay** [arendst/Tasmota#15514] [Q&A] [answered]
Hello all, 
I was wondering if there is any documentation on the settings exposed via the Sensor12 option listed here: 
https://tasmota.github.io/docs/Commands/#sensors

I understand the concept of single / differential however the documentation shows to enter D0 to D5 for either of the settings, with no indication of what those five states do. The datasheet for ADS1115 lists numerous possibilities but Tasmota doesn't indicate what is mapped to what. 

I've searched PR's as well as the normal sources and do not see this info anywhere - if it exists can I get a pointer to where? 
Either way, some elaboration in the docs would be great. 

Thank you! Tasmota is an amazing project <3

**Comment by barbudor** [comment, marked as answer]
As of today there is no more than PR #15001  from @FransOv

If I remember the discussion the command is  `Sensor12 mg`
Where `mg`  is a combination of mode and gain:
- m = S : single-ended mode, each channel 0, 1, 2, 3 match idependently each input from AIN0 to AIN3
- m = D :  differential mode, present only 2 channels where channel 0 is AIN0 - AIN1  and channel 1 in AIN2 - AIN3
- g sets the gain/range with
   - 0 : 6.144V full range
   - 1 : 4.096V full range
   - 2 : 2.048V full range
   - 3 : 1.024V full range
   - 4 : 0.512V full range
   - 5 : 0.256V full range
 
 For exemple `Sensor12 S2` set single-ended mode with 2.048V full range while `Sensor12 D1` set differential mode with 4.096V full range

**Reply by phinnay** [reply]
Thank you for the explanation!  This is helpful and what I needed to finish my project. Great work all around by group Tasmota :D

**Comment by DaBaBW** [comment]
Do I have to save the mode?
`Sensor12 S3` works fine for me, but after a restart it is always S0 again.

**Reply by sfromis** [reply]
It is not a persistent property, you need to set it after each boot, maybe using a rule triggering on `System#init`.

**Reply by Failx92** [reply]
Hi, 
same problem after restart is S0 again.

I tried the rule option with System#init and System#Boot but it doesnt work. 

The sensor has not yet been recognized when the trigger is activated and fail. 

Any hints?

**Reply by FransOv** [reply]
Works OK for me in a rule on system#boot

**Reply by sfromis** [reply]
Yeah, `System#boot` is triggered later than `System#init`

**Reply by Failx92** [reply]
Thx for reply,  I tried:
Rule1 System#Boot  DO Sensor12 S2 ENDON
Rule1 System#init  DO Sensor12 S2 ENDON

the rule has been confirmed in the console

I will send the console output later

**Reply by sfromis** [reply]
You did make sure to also enable the rule set?

**Reply by FransOv** [reply]
> Thx for reply, I tried: Rule1 System#Boot DO Sensor12 S2 ENDON Rule1 System#init DO Sensor12 S2 ENDON
> 
> the rule has been confirmed in the console
> 
> I will send the console output later

Command should be:
```
Rule1 on system#boot do sensor12 s2 endon
```

**Reply by sfromis** [reply]
In addition to fixing the bad command, also make sure to also enable the rule set
`Rule1 1`

**Reply by Failx92** [reply]
Thanks alot Mates, 

You are right @FransOv I miss the "on" after "Rule1" -.- so the rule was done after startup but not on the trigger and that confused me. 

Its fixed :)
